q2_feature_classifier/_vsearch.py

```diff
@@ -25,53 +31,161 @@ def classify_consensus_vsearch(query: DNAFASTAFormat,
+            _run_command(cmd)
+            sparse_reference = qiime2.Artifact.import_data(
+                'FeatureData[Sequence]', output.name)
+
+            # perform rough positive filter on query sequences
+            query, misses, = exclude(
+                query_sequences=query, reference_sequences=sparse_reference,
+                method='vsearch', perc_identity=perc_identity,
+                perc_query_aligned=query_cov, threads=threads)
+
+    # find exact matches, perform LCA consensus classification
+    taxa1, = ccv(query=query, reference_reads=reference_reads,
+                 reference_taxonomy=reference_taxonomy, maxaccepts=maxaccepts,
+                 strand=strand, min_consensus=min_consensus,
+                 search_exact=True, threads=threads)
+
+    # Annotate taxonomic assignments with classification method
+    taxa1 = _annotate_method(taxa1, 'VSEARCH')
+
+    # perform second pass classification on unassigned taxa
+    # filter out unassigned seqs
+    try:
+        query, = filter_seqs(sequences=query, taxonomy=taxa1,
+                             include=_get_default_unassignable_label())
+    except ValueError:
+        # get ValueError if all sequences are filtered out.
+        # so if no sequences are unassigned, return exact match results
+        return taxa1
+
+    # classify with sklearn classifier
+    taxa2, = cs(reads=query, classifier=classifier,
+                reads_per_batch=reads_per_batch, n_jobs=threads,
+                confidence=confidence, read_orientation=read_orientation)
+
+    # Annotate taxonomic assignments with classification method
+    taxa2 = _annotate_method(taxa2, 'sklearn')
+
+    # merge into one big happy result
+    taxa, = merge(data=[taxa2, taxa1])
+    return taxa
+
+
+output_descriptions = {
+    'classification': 'The resulting taxonomy classifications.'}
+
+parameters = {'maxaccepts': Int % Range(1, None) | Str % Choices(['all']),
+              'perc_identity': Float % Range(0.0, 1.0, inclusive_end=True),
+              'query_cov': Float % Range(0.0, 1.0, inclusive_end=True),
+              'strand': Str % Choices(['both', 'plus']),
+              'min_consensus': Float % Range(0.5, 1.0, inclusive_end=True,
+                                             inclusive_start=False),
+              'threads': Int}
+
+inputs = {'query': FeatureData[Sequence],
+          'reference_reads': FeatureData[Sequence],
+          'reference_taxonomy': FeatureData[Taxonomy]}
+
+input_descriptions = {'query': 'Sequences to classify taxonomically.',
+                      'reference_reads': 'reference sequences.',
+                      'reference_taxonomy': 'reference taxonomy labels.'}
+
+parameter_descriptions = {
+    'strand': 'Align against reference sequences in forward ("plus") '
+              'or both directions ("both").',
+    'maxaccepts': 'Maximum number of hits to keep for each query. Set to '
+                  '"all" to keep all hits > perc_identity similarity. Must '
```

Comment:
Just a general note, I think that communicating range constraints is generally a job for the interface. No worries on including it here (it can't hurt, and might help teach people how to read the type annotations!), but just wanted to toss that out there for future work.

Reply:
Yep, this is an old parameter description (Nick's first plugin method 👶) — I just moved these parameter descriptions out of the `classify_consensus_vsearch` registration without reading them!

I am happy to change now that we are shining some lights in the ol' gear box here.